Runner/JobManager.py

```diff
@@ -0,0 +1,380 @@
+            self.duration = (datetime.timedelta(seconds=(pool_time + job_time)))
+            self.check_expected_output(batch_service_client)
+
+    def retry(self, batch_service_client, blob_client, timeout):
+        """
+        Retries a job if it failed due to a NOT_COMPLETE or UNEXPECTED_OUTPUT. If the pool fails we don't retry the job
+        
+        :param timeout: How long we should wait for the job to complete
+        :param batch_service_client: A Batch service client.
+        :type batch_service_client: `azure.batch.BatchServiceClient`
+        :param blob_client: A blob service client.
+        :type blob_client: `azure.storage.blob.BlockBlobService`
+        """
+        if self.status.job_state == Utils.JobState.NOT_COMPLETE or self.status.job_state == Utils.JobState.UNEXPECTED_OUTPUT:
+            # Deletes the resources needed for the old job.
+            self.delete_resouces(batch_service_client, blob_client, True)
+            Logger.warn(
+                "Job [{}] did not complete in time so it will be recreated with the '-retry' postfix ".format(
+                    self.job_id))
+            # Set a new job id
+            self.job_id = self.job_id + "-retry"
+            self.upload_assets(blob_client)
+            self.create_and_submit_job(batch_service_client)
+            self.wait_for_job_results(batch_service_client, timeout)
+
+    def delete_pool(self, batch_service_client):
+        """
+        Deletes the pool the if the pool, if the pool has already been deleted or marked for deletion it
+        should ignore the batch exception that is thrown. These errors come up due to multiple jobs using the same pool
+        and when a the job cleans up after it's self it will call delete on the same pool since they are a shared resource.
+
+        :param batch_service_client: A Batch service client.
+        :type batch_service_client: `azure.batch.BatchServiceClient`
+        """
+        Logger.info("Deleting pool: {}.".format(self.pool_id))
+        try:
+            batch_service_client.pool.delete(self.pool_id)
+        except batchmodels.batch_error.BatchErrorException as batch_exception:
+            if Utils.expected_exception(
+                    batch_exception, "The specified pool has been marked for deletion"):
+                Logger.warn(
+                    "The specified pool [{}] has been marked for deletion.".format(
+                        self.pool_id))
+            elif Utils.expected_exception(batch_exception, "The specified pool does not exist"):
+                Logger.warn(
+                    "The specified pool [{}] has been deleted.".format(
+                        self.pool_id))
+            else:
+                print
+                traceback.print_exc()
+                Utils.print_batch_exception(batch_exception)
+
+    def delete_resouces(self, batch_service_client, blob_client, force_delete=False):
```

Comment:
🐛 name mistake I think you meant : `delete_resources` not `resouces`

Reply:
good spot!

Reply:
All good! @matthew-rollitt I will drop a check list before you I am done with this, please test your scenarios, also look at the PEP 8 recommendations above.